packages/dev/core/src/Meshes/greasedLineRibbonMesh.ts

```diff
@@ -0,0 +1,561 @@
+import type { Scene } from "../scene";
+import { Quaternion, Vector3 } from "../Maths/math.vector";
+import { Mesh } from "./mesh";
+import { Buffer } from "../Buffers/buffer";
+import type { Nullable } from "../types";
+import type { Node } from "../node";
+import { DeepCopier } from "../Misc/deepCopier";
+import { GreasedLineTools } from "../Misc/greasedLineTools";
+import type { GreasedLineMeshOptions, GreasedLineRibbonOptions } from "./greasedLineBaseMesh";
+import { GreasedLineBaseMesh, GreasedLineRibbonAutoDirectionMode, GreasedLineRibbonFacesMode, GreasedLineRibbonPointsMode } from "./greasedLineBaseMesh";
+
+Mesh._GreasedLineRibbonMeshParser = (parsedMesh: any, scene: Scene): Mesh => {
+    return GreasedLineRibbonMesh.Parse(parsedMesh, scene);
+};
+
+/**
+ * GreasedLine
+ */
+export class GreasedLineRibbonMesh extends GreasedLineBaseMesh {
+    /**
+     * Defaule line width
+     */
+    public static DEFAULT_WIDTH = 0.1;
+
+    private static _RightHandedForwardReadOnlyQuaternion = Quaternion.RotationAxis(Vector3.RightHandedForwardReadOnly, Math.PI / 2);
+    private static _LeftHandedForwardReadOnlyQuaternion = Quaternion.RotationAxis(Vector3.LeftHandedForwardReadOnly, Math.PI / 2);
+    private static _LeftReadOnlyQuaternion = Quaternion.RotationAxis(Vector3.LeftReadOnly, Math.PI / 2);
+
+    /**
+     * Direction which the line segment will be thickened if drawn on the XY plane
+     */
+    public static DIRECTION_XY = Vector3.LeftHandedForwardReadOnly; // doesn't matter in which handed system the scene operates
+    /**
+     * Direction which the line segment will be thickened if drawn on the XZ plane
+     */
+    public static DIRECTION_XZ = Vector3.UpReadOnly;
+    /**
+     * Direction which the line segment will be thickened if drawn on the YZ plane
+     */
+    public static DIRECTION_YZ = Vector3.LeftReadOnly;
+
+    private _paths: Vector3[][];
+    private _pathsOptions: { options: GreasedLineMeshOptions; pathCount: number }[];
+    private _vSegmentLengths: number[][];
+    private _uSegmentLengths: number[][];
+    private _vTotalLengths: number[];
+    private _uTotalLengths: number[];
+
+    private _counters: number[];
+    private _slopes: number[];
+    private _ribbonWidths: number[];
+
+    /**
+     * GreasedLineRibbonMesh
+     * @param name name of the mesh
+     * @param scene the scene
+     * @param _options mesh options

```

Comment:
Doc for `_pathOptions` missing. However, it seems this parameter is never used, so we can remove it?

Reply:
`_pathOptions` is used when parsing the mesh.